server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -785,6 +785,19 @@ public boolean hasIndex(Index index) {
         return indices.containsKey(index.getUUID());
     }
 
+    /**
+     * Returns a MapperService for the specified index if exists otherwise returns <code>null</code>.
+     */
+    @Nullable
+    public MapperService getMapperService(Index index) {
+        IndexService indexService = indexService(index);
+        if (indexService == null) {
+            return null;
+        } else {
+            return indexService.mapperService();
```

Comment:
Do we need to include small test for this new method? Also, for the change from index as string to index as object